substrate/frame/treasury/src/lib.rs

```diff
@@ -437,7 +448,8 @@ pub mod pallet {
 	impl<T: Config<I>, I: 'static> Hooks<BlockNumberFor<T>> for Pallet<T, I> {
 		/// ## Complexity
 		/// - `O(A)` where `A` is the number of approvals
-		fn on_initialize(n: frame_system::pallet_prelude::BlockNumberFor<T>) -> Weight {
+		fn on_initialize(_do_not_use_local_block_number: BlockNumberFor<T>) -> Weight {
+			let block_number = T::BlockNumberProvider::current_block_number();
```

Comment:
This might be a common error when people write pallets depending on a `BlockNumberProvider`

Reply:
Yea, `on_initialize` should be deprecated in favour of the new hooks from https://github.com/paritytech/polkadot-sdk/pull/1781.

Reply:
I am tempted to suggest removing the parameter from `Hooks` -- even without this change, it was superfluous as you (almost) always have access to block number from `frame_system` directly.